ui_framework/components/icon/delete_icon.js

```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import { KuiIcon } from './kui_icon';
+
+export function DeleteIcon() {
+  return <KuiIcon className="fa-trash"/>;
```

Comment:
@uboness - Sounds like there is a long history of snake_case vs CamelCase for file names in Kibana and snake_case eventually won out.  https://github.com/elastic/kibana/issues/5383

We would be breaking file name consistency in our code base if we use CamelCase for React Components.

/cc @Bargs @epixa @spalger